### ISSUE TITLE
Use mode=host for Nginx, to retrieve real IP behind request

### ISSUE THREAD

**Issue by vincelwt** [githubsaturn/captainduckduck#201]
So as specified in [this issue](https://github.com/moby/moby/issues/25526) it isn't possible to access the real ip address behind a request with Docker Swarm mode.

This can be very problematic, for example to limit specific pages to a certain IP or to do API rate limiting.

A workaround is to create the Nginx service with `mode=host`.
For example, using `docker service create` with 

```
--publish mode=host,published=80,target=80 \
--publish mode=host,published=443,target=443
```

I still don't understand fully the consequence of using this mode but it seems to cause some [limitations](https://github.com/moby/moby/issues/25526#issuecomment-360836426).

I don't know if this should be in Captain by default, but maybe as an option for those who need it? 

Thanks.

**Comment by tuefekci** [comment]
I am not that used to nginx but if i remeber correct they have a scripting language so you should be able to manipulate the request and add a custom header which you than parse in your application?

Its pritty late so i did not look into anything but nginx sits infront of the swarm if i see it so it should get propper ip rewrite it to another section in the request which should get served with the other request parameters and you should be able to get all the data from the client?

**Comment by vincelwt** [comment]
@tuefekci yes that's how you do it normally, but in this case the problem seems to be that Nginx doesn't itself have access to the IP.  
Checking the logs, all requests seem to come from the same IP (something like 10.255.0.2)

**Comment by githubsaturn** [comment]
Done in https://github.com/githubsaturn/captainduckduck/commit/4dae8da0d66f4d888e753e89c09928039992b723

Will be released in v0.7.0
